spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -241,6 +249,43 @@ public long getNextExecutionId(){
 		return taskIncrementer.nextLongValue();
 	}
 
+	@Override
+	public Long getTaskExecutionIdByJobExecutionId(long jobExecutionId) {
+		try {
+			return jdbcTemplate.queryForObject(
+					getQuery(FIND_TASK_EXECUTION_BY_JOB_EXECUTION_ID),
+					new Object[] { jobExecutionId },
+					Long.class);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public Set<Long> getJobExecutionIdsByTaskExecutionId(long taskExecutionId) {
+		try {
+			return jdbcTemplate.query(
+					getQuery(FIND_JOB_EXECUTION_BY_TASK_EXECUTION_ID),
+					new Object[] {taskExecutionId},
+					new ResultSetExtractor<Set<Long>>() {
+						@Override
```

Comment:
I didn't make it that because modifying the set doesn't modify the underlying data store...in the map case, it does.